docs/DEV.md

```diff
@@ -0,0 +1,179 @@
+
+## Loading the Web Extension in Firefox
+
+You can have Firefox automatically launched and the add-on installed by running:
+
+`$ npm run firefox`
+
+To load the extension manually instead, open (preferably) the [Developer Edition of Firefox](https://www.mozilla.org/firefox/developer/) and load the `.xpi` using the following steps:
+
+* Navigate to _about:config_ and set `extensions.legacy.enabled` to `true`. This permits the loading of the embedded Web Extension since new versions of Firefox are becoming restricted to pure Web Extensions only.
+* Navigate to _about:debugging_ in your URL bar
+* Select "Load Temporary Add-on"
+* Find and select the latest xpi file you just built.
+
+## Seeing the add-on in action
+
+To debug installation and loading of the add-on:
+
+* Navigate to _about:config_ and set `shield.testing.logging.level` to `10`. This permits shield-add-on log output in browser console
+* Open the Browser Console using Firefox's top menu at `Tools > Web Developer > Browser Console`. This will display Shield (loading/telemetry) and log output from the add-on.
+
+See [TESTPLAN.md](./TESTPLAN.md) for more details on how to see this add-on in action and hot it is expected to behave.
+
+## Automated launch of Firefox with add-on installed
+
+`$ npm run firefox` starts Firefox and automatically installs the add-on in a new profile and opens the browser console automatically.
+
+Note: This runs in a recently created profile. To have the study run despite the eligibility requirement of having at least 1 day old profiles, a config override is set in place to force the study to run.
+
+## Automated testing
+
+`npm run test` verifies the telemetry payload as expected at Firefox startup and add-on installation in a clean profile, then does **optimistic testing** of the _commonest path_ though the study for a user
+
+* prove the notification bar ui opens
+* _clicking on the left-most button presented_.
+* verifying that sent Telemetry is correct.
+
+Code at [/test/functional_test.js](/test/functional_test.js).
+
+Note: The functional tests are using async/await, so make sure you are running Node 7.6+
+
+The functional testing set-up is imported from [https://github.com/mozilla/share-button-study]() which contains plenty of examples of functional tests relevant to Shield study addons.
+
+## Watch
+
+You can automatically build recent changes and package them into a `.xpi` by running the following from the top level directory:
+
+`$ npm run watch`
+
+Now, anytime a file is changed and saved, node will repackage the add-on. You must reload the add-on as before, or by clicking the "Reload" under the add-on in _about:debugging_. Note that a hard re-load is recommended to clear local storage. To do this, simply remove the add-on and reload as before.
+
+Note: This is currently only useful if you load the extension manually - it has no effect when running `npm run firefox`.
+
+## Directory Structure and Files
```

Comment:
This file tree may need a refresh. For example, you removed `about.md`, `survival.md` and `tutorial.md` files, but they're still listed here.

Reply:
Very true. Done